modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -148,6 +149,27 @@ class SliceLayerImpl : public SliceLayer
             const DictValue &sizesOrEnds = params.has("size") ? params.get("size") : params.get("end");
             CV_Assert(begins.size() == sizesOrEnds.size());
 
+            if (params.has("steps"))
+            {
+                const DictValue &steps = params.get("steps");
+                sliceSteps.resize(1);
+                sliceSteps[0].resize(steps.size());
+
+                for (int i = 0; i < steps.size(); ++i)
+                {
+                    int step = steps.get<int>(i);
+                    CV_Assert(step != 0);
+                    if (step < 0)
+                    {
+                        hasNegSteps = true;
+                        neg_step_dims.push_back(i);
```

Comment:
We can use the fact that neg_step_dims is not empty as an indicator that `hasNegSteps` is `true`